### ISSUE TITLE
stage2  VTG loss wrong ?

### ISSUE THREAD

**Issue by WorldHellooo** [OpenGVLab/Ask-Anything#124]
inputs_embeds = torch.cat([bos_embeds, img_embeds, to_regress_embeds], dim=1) 
you just concat all text emdeds with img_embeds? But BLIP2 concat half text emdeds with img_embeds.

This is from blip2:  "For encoder-decoder-based LLMs, we pre-train with the prefix language modeling loss, where we split a text into two parts. The prefix text is concatenated with the visual representation as input to the LLM’s encoder."

**Comment by Andy1621** [comment]
Thanks for your question. 
BLIP2 only concats half text embeds with img_embeds when using encoder-decoder-based LLMs like FLAN-T5. 
For those decoded-only models like LLaMA, it's natural to concats all the text embeds that have been masked, i.e., `to_regress_embeds `. For training, the model should regress those text embeds.

**Comment by yinanhe** [comment]
Hi, we will close this issue.

Feel free to contact us if you have other questions.
